pkg/testing/fixture_install.go

```diff
@@ -128,23 +133,28 @@ func (f *Fixture) Install(ctx context.Context, installOpts *InstallOpts, opts ..
 
 		// environment variable AGENT_KEEP_INSTALLED=true will skip the uninstall
 		// useful to debug the issue with the Elastic Agent
-		if keepInstalled() {
+		if keepInstalled() && f.t.Failed() {
```

Comment:
What is the motivation to change this behavior? Previously we would skip uninstall if the `AGENT_KEEP_INSTALLED` flag was set to `true`.  Now we will skip uninstall if the `AGENT_KEEP_INSTALLED` flag was set to `true` and the test failed as well.

Reply:
Never mind, I see the explanation the `else` log line 🙂.